src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -2913,16 +2915,18 @@ static void window_ride_construction_update_widgets(rct_window* w)
         {
             // Enable helix
             window_ride_construction_widgets[WIDX_SLOPE_DOWN_STEEP].type = WindowWidgetType::FlatBtn;
-            if (rideType != RIDE_TYPE_SPLASH_BOATS)
-                window_ride_construction_widgets[WIDX_SLOPE_UP_STEEP].type = WindowWidgetType::FlatBtn;
+            window_ride_construction_widgets[WIDX_SLOPE_UP_STEEP].type = WindowWidgetType::FlatBtn;
```

Comment:
Why did you change these?

Reply:
It was hard for me to come up with the best way to not inadvertently disable the steep slope up for other rides. I was trying to get the steep slope up button enabled for Splash Boats only when the show all drawable track pieces cheat is enabled. It will still be disabled when that cheat is turned off. I just could not easily determine the best way to code this without disabling it for all other rides.

Reply:
Unless you swapped around enabled/disabled around by accident, I cannot make much sense of your answer:

> I was trying to get the steep slope up button enabled for Splash Boats

Why would you want to enable the steep slope up for Splash Boats anyway? The up pieces have rollers and look distinct from the down pieces, and I don't remember a sprite existing for a steep up piece.

Reply:
During my testing, I learned that the Splash Boat vehicles can actually climb up the steep slope up pieces (the River Rafts vehicles can also traverse slopes, both up and down). Steep slope up reuses the same sprites as those for steep slope down, but it apparently works.

Reply:
The track lacks them, which was my point:

![afbeelding](https://user-images.githubusercontent.com/1478678/102699224-6df5cf00-4243-11eb-85cb-e1b4da31fc82.png)

The cheat is there to enable _drawable_ track pieces. I do not count this as such. Please revert the changes to the Splash Boats and this bit of code.

Reply:
Okay then. Before I complete this, should I also apply these changes to the River Rafts, or leave them as-is? If I revert this code, the steep slopes will become accessible to the River Rafts, but not Splash Boats.

Reply:
Restore the old code, but add an extension for River Rafts.

In the longer run, I want to introduce a flag that says the ride uses a water channel, which can be checked instead of doing a ride type check. You can also do that in this PR if you feel like trying that.

Reply:
> In the longer run, I want to introduce a flag that says the ride uses a water channel, which can be checked instead of doing a ride type check. You can also do that in this PR if you feel like trying that.
Cool, I think that might be a better fit. First, I need to know how to identify a ride as having a water channel.
Side note: OpenGL rendering is causing a bunch of problem for me:
```
openrct2 $ openrct2: /home/richard/OpenRCT2/src/openrct2-ui/drawing/engines/opengl/TextureCache.h:116: AtlasTextureInfo Atlas::Allocate(int32_t, int32_t): Assertion `!_freeSlots.empty()' failed.
                    Aborted (core dumped)
```
Should I revert to an earlier commit to get around this problem, or use software rendering for now? I tried that, but the font looks all messed up! I know that other people have reported this problem.

Reply:
That was a bad commit that was merged on Friday morning and fixed in the afternoon. I have already rebased your branch against the most recent commit, so you should not have any problems. If you still have, make sure you rebuild g2.dat.

Rides with water channels are Log Flume, Splash Boats and River Rafts.

Reply:
I think I have this issue fixed now; however identifying which rides are using water channels looks like to be out of scope for what I want to merge here. It may appear in another pull request in the future.